x/gov/handler_sidechain.go

```diff
@@ -6,6 +6,20 @@ import (
 )
 
 func handleMsgSideChainSubmitProposal(ctx sdk.Context, keeper Keeper, msg MsgSideChainSubmitProposal) sdk.Result {
+	if sdk.IsUpgrade(sdk.BCFusionSecondHardFork) {
+		return sdk.ErrMsgNotSupported("").Result()
+	}
+	if sdk.IsUpgrade(sdk.BCFusionFirstHardFork) {
+		ctx, err := keeper.ScKeeper.PrepareCtxForSideChain(ctx, msg.SideChainId)
```

Comment:
just move `ctx, err := keeper.ScKeeper.PrepareCtxForSideChain(ctx, msg.SideChainId)` outside the condition.